### PR TITLE
SC-2004 - Added implementation for backward compatibility

### PR DESCRIPTION
**Pull request by rheafernandes**

### CODE REVIEW

File: content-api/content-service/app/controllers/BaseController.scala [hunk 1]
```diff
@@ -68,16 +68,19 @@ abstract class BaseController(protected val cc: ControllerComponents)(implicit e
...
     }
 
-    def getResult(apiId: String, actor: ActorRef, request: org.sunbird.common.dto.Request) : Future[Result] = {
+    def getResult(apiId: String, actor: ActorRef, request: org.sunbird.common.dto.Request, categoryMapping: Boolean = false) : Future[Result] = {
         val future = Patterns.ask(actor, request, 30000) recoverWith {case e: Exception => Future(ResponseHandler.getErrorResponse(e))}
         future.map(f => {
             val result = f.asInstanceOf[Response]
             result.setId(apiId)
             setResponseEnvelope(result)
+            if (categoryMapping) setContentAndCategoryTypes(result.getResult.getOrDefault("content", new util.HashMap[String, AnyRef]()).asInstanceOf[util.Map[String, AnyRef]])
```
Comment (PradyumnaNagendra) [inline content-api/content-service/app/controllers/BaseController.scala:83]: Can we do this for ok response only, not for all the responses.

File: content-api/content-service/app/controllers/BaseController.scala [hunk 2]
```diff
@@ -105,4 +108,26 @@ abstract class BaseController(protected val cc: ControllerComponents)(implicit e
         request.setObjectType(objectType);
         request.setContext(contextMap)
     }
+
+    private def setContentAndCategoryTypes(input: java.util.Map[String, AnyRef]): Unit = {
+        val contentType = input.get("contentType").asInstanceOf[String]
+        val primaryCategory = input.get("primaryCategory").asInstanceOf[String]
+        val categoryMap: java.util.Map[String, AnyRef] = Platform.getAnyRef("contentTypeToPrimaryCategory",
```
Comment (PradyumnaNagendra) [inline content-api/content-service/app/controllers/BaseController.scala:115]: move reading from config to class level

**Review by PradyumnaNagendra — CHANGES_REQUESTED** [review]

File: content-api/content-service/app/controllers/BaseController.scala (hunk 1 above)
Comment (rheafernandes) [reply content-api/content-service/app/controllers/BaseController.scala:83]: Done

File: content-api/content-service/app/controllers/BaseController.scala (hunk 2 above)
Comment (rheafernandes) [reply content-api/content-service/app/controllers/BaseController.scala:115]: done

**Review by PradyumnaNagendra — APPROVED** [review]

**Review by maheshkumargangula — APPROVED** [review]

**Pull request merged**